### PR TITLE
Remove node auth

### PR DESCRIPTION
**Pull request by grumbach**
Remove signatures from node messages
- remove `msg_authority` from `MsgType::Node`
- remove the `NodeMsgAuthority` type altogether

### CODE REVIEW

File: sn_interface/src/messaging/serialisation/wire_msg.rs [hunk 1]
```diff
@@ -251,17 +251,14 @@ impl WireMsg {
                     msg,
                 })
             }
-            AuthKind::Node(node_signed) => {
+            // NB TODO remove this enum
+            AuthKind::Node(_) => {
```
Comment (dirvine) [inline sn_interface/src/messaging/serialisation/wire_msg.rs:255]: I thnk this should be the next PR when this one is merged.

**Review by dirvine — APPROVED** [review]
This code now reads significantly easier.  Looks good to me

File: sn_node/src/node/messaging/anti_entropy.rs [hunk 2]
```diff
@@ -483,51 +482,6 @@ mod tests {
             .await
     }
 
-    #[tokio::test]
```
Comment (joshuef) [inline sn_node/src/node/messaging/anti_entropy.rs:486]: @grumbach is this test not still valid? It's all about the SAP proof chain,as opposed to signed msgs

File: sn_interface/src/messaging/serialisation/wire_msg.rs (hunk 1 above)
Comment (grumbach) [reply sn_interface/src/messaging/serialisation/wire_msg.rs:255]: Yes that was the plan!

File: sn_node/src/node/messaging/anti_entropy.rs (hunk 2 above)
Comment (grumbach) [reply sn_node/src/node/messaging/anti_entropy.rs:486]: It's testing a function that doesn't exist anymore, testing the key in the msg auth. 
Similar checks are also in the update function for section tree, so it was redundant.

**Pull request merged**